rest-service/manager_rest/resource_manager.py

```diff
@@ -379,17 +379,43 @@ def publish_blueprint(self,
         )
         return self.sm.put(new_blueprint)
 
-    def delete_blueprint(self, blueprint_id):
+    def delete_blueprint(self, blueprint_id, force):
         blueprint = self.sm.get(models.Blueprint, blueprint_id)
         self.validate_modification_permitted(blueprint)
 
+        if not force:
+            used_blueprints = list(set(
+                d.blueprint_id for d in
+                self.sm.list(models.Deployment, include=['blueprint_id'])))
+            imported_blueprints = [b.plan[constants.IMPORTED]
+                                   for b in self.sm.list(models.Blueprint,
+                                                         include=['plan'],
+                                                         filters={'id': used_blueprints})]
+            if blueprint_id in imported_blueprints:
+                raise manager_exceptions.BlueprintInUseError(
+                    'Blueprint {} is currently in use. You can "force" '
+                    'blueprint removal.'.format(blueprint_id))
+
         if len(blueprint.deployments) > 0:
             raise manager_exceptions.DependentExistsError(
                 "Can't delete blueprint {0} - There exist "
                 "deployments for this blueprint; Deployments ids: {1}"
                 .format(blueprint_id,
                         ','.join([dep.id for dep
                                   in blueprint.deployments])))
+        # Delete blueprint resources from file server
```

Comment:
Why did you move it here?

Reply:
So I should not need to copy this in the V3.1 and that's the more relevant place (like the plugins)